core/src/main/java/org/elasticsearch/index/mapper/geo/GeoPointFieldMapper.java

```diff
@@ -245,25 +261,68 @@ public GeoPointFieldMapper build(BuilderContext context) {
+                    if (builder.bwc) {
+                        if (!builder.fieldType().ignoreMalformed) {
+                            builder.fieldType().ignoreMalformed = !XContentMapValues.nodeBooleanValue(fieldNode);
+                        }
+                    } else {
+                        this.throwDeprecationException(fieldName, Names.IGNORE_MALFORMED);
+                    }
+                   iterator.remove();
                 } else if (fieldName.equals("validate_lon")) {
-                    builder.fieldType().setValidateLon(XContentMapValues.nodeBooleanValue(fieldNode));
+                    if (builder.bwc) {
+                        if (!builder.fieldType().ignoreMalformed()) {
+                            builder.fieldType().ignoreMalformed = !XContentMapValues.nodeBooleanValue(fieldNode);
+                        }
+                    } else {
+                        this.throwDeprecationException(fieldName, Names.IGNORE_MALFORMED);
+                    }
                     iterator.remove();
                 } else if (fieldName.equals("validate_lat")) {
-                    builder.fieldType().setValidateLat(XContentMapValues.nodeBooleanValue(fieldNode));
+                    if (builder.bwc) {
+                        if (!builder.fieldType().ignoreMalformed) {
+                            builder.fieldType().ignoreMalformed = !XContentMapValues.nodeBooleanValue(fieldNode);
+                        }
+                    } else {
+                        this.throwDeprecationException(fieldName, Names.IGNORE_MALFORMED);
+                    }
+                    iterator.remove();
+                } else if (fieldName.equals(Names.COERCE)) {
+                    builder.fieldType().coerce = XContentMapValues.nodeBooleanValue(fieldNode);
+                    builder.fieldType().normalizeLat = builder.fieldType().coerce;
+                    builder.fieldType().normalizeLon = builder.fieldType().coerce;
                     iterator.remove();
                 } else if (fieldName.equals("normalize")) {
-                    builder.fieldType().setNormalizeLat(XContentMapValues.nodeBooleanValue(fieldNode));
-                    builder.fieldType().setNormalizeLon(XContentMapValues.nodeBooleanValue(fieldNode));
+                    if (builder.bwc) {
+                        builder.fieldType().coerce = XContentMapValues.nodeBooleanValue(fieldNode);
+                        builder.fieldType().setNormalizeLat(builder.fieldType().coerce);
+                        builder.fieldType().setNormalizeLon(builder.fieldType().coerce);
+                    } else {
+                        this.throwDeprecationException(fieldName, Names.COERCE);
+                    }
                     iterator.remove();
                 } else if (fieldName.equals("normalize_lat")) {
-                    builder.fieldType().setNormalizeLat(XContentMapValues.nodeBooleanValue(fieldNode));
+                    if (builder.bwc) {
```

Comment:
You could also just name it, as a local, so that the name implies what version is being checked for, but I would personally just put the version check directly here.